### PR TITLE
Prettified codebase

### LINKED ISSUE #324: Prettifying the codebase
The codebase is filled with inconsistent code styles which results in large diffs for small changes. This is unideal because it distracts from actual changes being made in PRs and is ugly and hard to look at. I suggest that we try and complete the following tasks for this issue:

- [ ] Find a way to specify formatting for the Hugo codebase ([prettier](https://discourse.gohugo.io/t/hugo-and-prettier/22596/3) or something else)
- [ ] Update DOCS to specify how the user can format their code before making PRs
- [ ] Create a GitHub Action that formats PRs or blocks them using a failed test (in case the PR owner doesn't format their code)
- [ ] Reformat all code in the codebase

### PR DESCRIPTION
**Pull request by AnimeAllstar**
- setup prettier config file for the repo
- ran prettier on the entire repo
- updated docs
- added GitHub Action that creates a commit on PRs to master if the code isn't prettified. (We could do this in the same commit, but I thought it'd be better to have a different commit). This happens before the merge when all the deploy preview tests are running. If you add a second commit with bad formatting to a live PR, it'll make another commit to the PR formatting your code.

You will have to run `yarn install` when you pull changes to run prettier locally.

Let me know what y'all think. We can definitely update the prettier config if there are any changes that should be made.

closes #324

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
+
+      - name: Prettify code
+        uses: creyD/prettier_action@v4.2
+        with:
+          prettier_options: --write **/*.+(js|html|css|json|md|yaml) --config .prettierrc
+          only_changed: True
+          prettier_plugins: prettier-plugin-go-template
+          commit_message: "prettified code"
```
Comment (kewbish) [inline .github/workflows/main.yml:24]: Nit: capitalize? (Also, the action provides a default commit message of `Prettified Code!`, so we could also leave this out.)

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
+
+      - name: Prettify code
+        uses: creyD/prettier_action@v4.2
+        with:
+          prettier_options: --write **/*.+(js|html|css|json|md|yaml) --config .prettierrc
```
Comment (kewbish) [inline .github/workflows/main.yml:21]: Nit: I think by default Prettier will by default [look at `.prettierrc`](https://prettier.io/docs/en/configuration.html) in this repo, so `--config .prettierrc` might not be neccessary?

File: .nvmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+18.12.1
```
Comment (kewbish) [inline .nvmrc:1]: Was this intended to be commited? If so, what is this required by (Prettier, GH Workflow)?

File: README.md [hunk 4]
```diff
@@ -2,10 +2,29 @@
...
+
+# start the development server
+hugo server
+```
+
 ## Contributing
 
-We welcome contributions of any kind including pages, suggestions, bug reports,
-pull requests etc. We would love to hear from you.
+We welcome contributions of any kind including pages, suggestions, bug reports, pull requests etc. We would love to hear from you.
```
Comment (kewbish) [inline README.md:27]: Could you add a line here along the lines of:

When you submit a PR, we automatically run the [Prettier](https://prettier.io/) code formatter with the [prettier-plugin-go-template](https://github.com/NiklasPor/prettier-plugin-go-template) plugin to help with Go template formatting. Please ensure you checkout the latest version of your PR after any formatting commits have been made when continuing to work on your PR.

just so people aren't confused as to why a commit is automatically made + they also won't run into issues if we request changes?

**Review by kewbish — APPROVED** [review]
We have a bit of a chicken and egg situation here: I can't run the GitHub action on this repo if it hasn't been merged here, but I saw on your personal fork that you've tested it and it works, so I'm comfortable with approving.

This should be revertible if things go wrong (and if they go wrong, Netlify won't build the latest version and will continue to serve the old version) so we should be safe. Netlify successfully built this PR, so there shouldn't be any issues.

I left a couple nits below, feel free to address or leave them.

Thanks so much for taking this on!

File: .github/workflows/main.yml (hunk 1 above)
Comment (AnimeAllstar) [reply .github/workflows/main.yml]: I didn't like the exclamation point. I've changed it to `Prettified code`.

File: .github/workflows/main.yml (hunk 2 above)
Comment (AnimeAllstar) [reply .github/workflows/main.yml]: You're right, I just tested this. I took a look at [this](https://github.com/creyD/prettier_action/issues/41#issuecomment-787981694) and decided to add it.

File: README.md (hunk 4 above)
Comment (AnimeAllstar) [reply README.md:27]: Cool, do we want to create a new commit or amend the commmit instead? @kewbish

File: .nvmrc (hunk 3 above)
Comment (AnimeAllstar) [reply .nvmrc:1]: I've added some docs regarding this in https://github.com/ubccsss/ubccsss.org/pull/348/commits/6897aede1f15120289a8d863c52750f8194687fc. Let me know if it looks good. @kewbish

File: README.md (hunk 4 above)
Comment (kewbish) [reply README.md:31]: Creating a new commit will be cleaner IMO because contributers can simply `git pull` instead.

File: .nvmrc (hunk 3 above)
Comment (kewbish) [reply .nvmrc:1]: This makes sense, looks great! Thank you.

**Pull request merged**